soh/src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -209,12 +216,129 @@ void func_809DF730(EnCow* this, GlobalContext* globalCtx) {
     }
 }
 
+struct CowInfo {
+    int cowId;
+    RandomizerCheck randomizerCheck;
+};
+
+struct CowInfo EnCow_GetCowInfo(EnCow* this, GlobalContext* globalCtx) {
+    s32 uniqueCoords = this->actor.world.pos.x + this->actor.world.pos.z;
+    struct CowInfo cowInfo;
+
+    switch (globalCtx->sceneNum) {
+        case SCENE_SOUKO: // Lon Lon Tower
+            switch (uniqueCoords) {
+                // Two cases here cause this cow is moved in randomizer
+                case -173:
+                case -72:
+                    cowInfo.cowId = 0;
+                    cowInfo.randomizerCheck = RC_LLR_TOWER_LEFT_COW;
+                    break;
+                default:
+                    cowInfo.cowId = 1;
+                    cowInfo.randomizerCheck = RC_LLR_TOWER_RIGHT_COW;
+                    break;
+            }
+            break;
+        case SCENE_MALON_STABLE:
+            switch (uniqueCoords) {
+                // Two cases here cause this cow is moved in randomizer
+                case -257:
+                case -138:
+                    cowInfo.cowId = 2;
+                    cowInfo.randomizerCheck = RC_LLR_STABLES_RIGHT_COW;
+                    break;
+                default:
+                    cowInfo.cowId = 3;
+                    cowInfo.randomizerCheck = RC_LLR_STABLES_LEFT_COW;
+                    break;
+            }
+            break;
+        case SCENE_KAKUSIANA: // Grotto
+            switch (uniqueCoords) {
+                case 1973:
+                    cowInfo.cowId = 4;
+                    cowInfo.randomizerCheck = RC_DMT_COW_GROTTO_COW;
+                    break;
+                default:
+                    cowInfo.cowId = 5;
+                    cowInfo.randomizerCheck = RC_HF_COW_GROTTO_COW;
+                    break;
+            }
+            break;
+        case SCENE_LINK_HOME:
+            cowInfo.cowId = 6;
+            cowInfo.randomizerCheck = RC_KF_LINKS_HOUSE_COW;
+            break;
+        case SCENE_LABO: // Impa's house
+            cowInfo.cowId = 7;
+            cowInfo.randomizerCheck = RC_KAK_IMPAS_HOUSE_COW;
+            break;
+        case SCENE_SPOT09: // Gerudo Valley
+            cowInfo.cowId = 8;
+            cowInfo.randomizerCheck = RC_GV_COW;
+            break;
+        case SCENE_SPOT08: // Jabu's Belly
+            cowInfo.cowId = 9;
+            cowInfo.randomizerCheck = RC_JABU_JABUS_BELLY_MQ_COW;
+            break;
+    }
+
+    return cowInfo;
+}
+
+void EnCow_MoveCowsForRandomizer(EnCow* this, GlobalContext* globalCtx) {
+    struct CowInfo cowInfo = EnCow_GetCowInfo(this, globalCtx);
```

Comment:
since there are only 2 cows being moved, i think it makes more sense to just keep the pre-moved cow identifying in here instead